src/logic/embedding.lean

```diff
@@ -100,6 +110,18 @@ protected def some {α} : α ↪ option α :=
 def subtype {α} (p : α → Prop) : subtype p ↪ α :=
 ⟨subtype.val, λ _ _, subtype.eq'⟩
 
+/-- Choosing an element `b : β` gives an embedding of `punit` into `β`. -/
+def punit {β : Sort*} (b : β) : punit ↪ β :=
+⟨λ _, b, by { rintros ⟨⟩ ⟨⟩ _, refl, }⟩
+
+/-- Fixing an element `b : β` gives an embedding `α ↪ α × β`. -/
+def inl (α : Sort*) {β : Sort*} (b : β) : α ↪ α × β :=
```

Comment:
I would expect `inl` and `inr` to be the embeddings into the sum types. So maybe we can cook up new names for these ones? I don't have a good suggestion atm, sorry.

Reply:
Yes, I agree the naming is awkward here. In the category theory files the corresponding functors into the product category are called `inl` and `inr`, so at least there's precedent, even if it's bad.

Reply:
Oh dear, and the functors into the `sum` category are `inl_` and `inr_`...

Reply:
Hmm... would it make sense to swap those names? The map into the sum is more canonical then the one for prods, and besides that `inl_ x` can be read as having `x` as subscript of `inl_`, like we would write `$$\iota_x$$ in latex.

Reply:
I think we should just find a completely different name for these `α ↪ α × β` maps (not sure where that arrow came from!) and we can rename the category theory ones at the same time.

Reply:
The same name is used in `monoid_hom.inl`/`monoid_hom.inr` and `linear_map.inl`/`linear_map.inr` though there is no `monoid`/`module` structure on `sum α β`.

Reply:
Or something with `fiber` in the name? `fiber_fst` and `fiber_snd`?

Reply:
I would think that for `x : X` and `Y` something like `fiber_fst x Y` would be the (sub)type `{x} \times Y`...

Reply:
`sectl` and `sectr`? Unscrutable and short just like `inl` and `inr`.

Reply:
Sounds good!

Reply:
Okay, `sectl` and `sectr` it is. I've renamed the category theory versions as well.

Reply:
Do you also want to rename `inl_` and `inr_` to `inl` and `inr`? Or not in this PR?